src/main/java/com/onarandombox/multiverseinventories/share/Sharables.java

```diff
@@ -582,6 +590,149 @@ public boolean updatePlayer(Player player, PlayerProfile profile) {
             }).serializer(new ProfileEntry(false, "potions"), new PotionEffectSerializer())
             .altName("potion").altName("potions").build();
 
+    /**
+     * Sharing Advancements.
+     */
+    public static final Sharable<List> ADVANCEMENTS = new Sharable.Builder<List>("advancements", List.class,
+            new SharableHandler<List>() {
+                @Override
+                public void updateProfile(PlayerProfile profile, Player player) {
+                    Set<String> completedAdvancements = new HashSet<>();
```

Comment:
Why use a HashSet if you are going to cast it into a List anyways?

Reply:
The reason for using HashSet is that Sets can't have duplicate values and the complexity of HashSet's `contains` method is O(1).

The reason for casting it to a List is because there is a built in serializer for Lists, so we don't have to write one for Sets.

Reply:
> The reason for using HashSet is that Sets can't have duplicate values and the complexity of HashSet's `contains` method is O(1).

The thing is you didn't use the `contains` method for completedAdvancements variable tho. Unless I missed it.

Reply:
oh wait, yea duplicated values mb